src/Workleap.Extensions.Mongo.Tests/MongoCollectionBuilderTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver;
+
+namespace Workleap.Extensions.Mongo.Tests;
+
+public sealed class MongoCollectionBuilderTests
+{
+    private readonly MongoCollectionBuilder<TestDocument> _builder = new();
+
+    [Fact]
+    public async Task Builder_Builds_Correctly()
+    {
+        var collectionName = "collection1";
+
+        Action<BsonClassMap<TestDocument>> classMapInitializer = map =>
+        {
+            map.MapIdProperty(x => x.Id);
+        };
+        
+        this._builder.CollectionName(collectionName)
```

Comment:
Should we check + test that the collection name is not null or empty?

Reply:
We already have a check when calling AddConfigurations, and the build happens the line just before. I dont think its worth adding it there too. I feel like all validatons should be together during AddConfigurations